scripts/src/java/org/oppia/android/scripts/coverage/CoverageReporter.kt

```diff
@@ -0,0 +1,198 @@
+ */
+class CoverageReporter(
+  private val repoRoot: String,
+  private val coverageReportList: List<CoverageReport>,
+  private val reportFormat: ReportFormat,
+) {
+  private val computedCoverageRatio = computeCoverageRatio()
+  private val formattedCoveragePercentage = "%.2f".format(computedCoverageRatio * 100)
+
+  private val filePath = coverageReportList.firstOrNull()?.filePath ?: "Unknown"
+
+  private val totalLinesFound = coverageReportList.getOrNull(0)?.linesFound ?: 0
+  private val totalLinesHit = coverageReportList.getOrNull(0)?.linesHit ?: 0
+
+  /**
+   * Generates a rich text report for the analysed coverage data based on the specified format.
+   * It supports Markdown and HTML formats.
+   *
+   * @return a pair where the first value is the computed coverage ratio represented in [0, 1]
+   *     and the second value is the generated report text
+   */
+  fun generateRichTextReport(): Pair<Float, String> {
+    println("report format: $reportFormat")
+    return when (reportFormat) {
+      ReportFormat.MARKDOWN -> generateMarkdownReport()
+      ReportFormat.HTML -> generateHtmlReport()
+    }
+  }
+
+  private fun generateMarkdownReport(): Pair<Float, String> {
+    val markdownContent =
+      """
+        ## Coverage Report
+
+        - **Covered File:** $filePath
+        - **Coverage percentage:** $formattedCoveragePercentage% covered
+        - **Line coverage:** $totalLinesHit / $totalLinesFound lines covered
+      """.trimIndent()
+
+    println("\n$markdownContent")
+
+    return Pair(computedCoverageRatio, markdownContent)
+  }
+
+  private fun generateHtmlReport(): Pair<Float, String> {
+    var htmlContent =
+      """
+    <!DOCTYPE html>
+    <html lang="en">
+    <head>
+      <meta charset="UTF-8">
+      <meta name="viewport" content="width=device-width, initial-scale=1.0">
+      <title>Coverage Report</title>
+      <style>
+        body {
+            font-family: Arial, sans-serif;
+            line-height: 1.6;
+            padding: 20px;
+        }
+        table {
+            width: 100%;
+            border-collapse: collapse;
+            margin-bottom: 20px;
+        }
+        th, td {
+            padding: 8px;
+            text-align: left;
+            border-bottom: 1px solid #fdfdfd;
+        }
+        .line-number-col {
+            width: 5%;
+        }
+        .source-code-col {
+            width: 95%;
+        }
+        .covered-line, .not-covered-line, .uncovered-line {
+            white-space: pre-wrap;
+            word-wrap: break-word;
+            box-sizing: border-box;
+            border-radius: 4px;
+            padding: 2px 8px 2px 4px;
+            display: inline-block;
+        }
+        .covered-line {
+            background-color: #c8e6c9; /* Light green */
+        }
+        .not-covered-line {
+            background-color: #ffcdd2; /* Light red */
+        }
+        .uncovered-line {
+            background-color: #fafafa; /* Half white */
```

Comment:
Given that this file renders in light mode, the white is barely visible and it took me a really long time to notice. Maybe change to something like light gray or ligh yellow.

Reply:
Thanks Adhiambo, so the uncovered lines are the ones represented in the kind of half white `#fafafa`. They basically didn't require any highlighting (lcov genhtml reports leave them as plain lines), I thought maybe it would look nice to separate lines of code from the background so added that. But as said it's barely visible because I didn't want to really make it any differentiable but still give a slight elevation. 

I tried adding a light gray `#F1F1F1`, I think that looks like a better way to represent as you said too. (Yellow couldn't be possibly added as it is something we can reserve for partially covered data). Also, I think the top summary possibly needed some design changes too so just retouched on few things. Can you check if this looks decent? 
_(Definitely needs a lot of re styling to look good but just wanted to keep it simple and usable for now)_

![Screenshot (1424)](https://github.com/oppia/oppia-android/assets/122200035/843ec397-f216-4079-b28f-a888cebef8b2)

Reply:
Light gray loooks ok. Half white was not visible, so it looked a bit like northing had happened to those lines. Gray gives the impression that they were checked, just uncovered.

Reply:
Ah got it! Thanks for the suggestion.